Source/santasyncservice/SNTPushClientNATS.mm

```diff
@@ -543,6 +582,171 @@ static void messageHandler(natsConnection *nc, natsSubscription *sub, natsMsg *m
+      response.set_result(santa::commands::v1::COMMAND_ERROR);
+      response.set_output("Unknown command type");
+    }
+
+    // Serialize and publish the response
+    // Serialization failures are logged but don't crash the client
+    std::string responseData;
+    if (!response.SerializeToString(&responseData)) {
+      LOGE(@"NATS: Failed to serialize command response (non-fatal)");
+      natsMsg_Destroy(msg);
+      return;
+    }
+
+    // Publish response asynchronously - failures are logged but don't crash
+    dispatch_async(self.connectionQueue, ^{
+      if (![self isConnectionAlive]) {
+        LOGW(@"NATS: Cannot send command response - not connected (non-fatal)");
+        return;
+      }
+
+      natsStatus status =
+          natsConnection_Publish(self.conn, [replyTopic UTF8String], responseData.data(),
+                                 static_cast<int>(responseData.length()));
+      if (status != NATS_OK) {
+        LOGE(@"NATS: Failed to publish command response to %@: %s (non-fatal)", replyTopic,
+             natsStatus_GetText(status));
+      } else {
+        LOGD(@"NATS: Sent command response to %@ (result: %d)", replyTopic, response.result());
+      }
+    });
+
+    natsMsg_Destroy(msg);
+  });
+}
+
+- (void)publishCommandResponse:(santa::commands::v1::SantaCommandResponseCode)resultCode
+                        output:(NSString *)output
+                  toReplyTopic:(NSString *)replyTopic {
+  // Failures are logged but don't crash the client
+  if (!replyTopic || replyTopic.length == 0) {
+    LOGW(@"NATS: Cannot publish command response - no reply topic provided (non-fatal)");
+    return;
+  }
+
+  santa::commands::v1::SantaCommandResponse response;
+  response.set_result(resultCode);
+  if (output) {
+    response.set_output([output UTF8String]);
+  }
+
+  std::string responseData;
+  if (!response.SerializeToString(&responseData)) {
+    LOGE(@"NATS: Failed to serialize command response (non-fatal)");
+    return;
+  }
+
+  dispatch_async(self.connectionQueue, ^{
+    if (![self isConnectionAlive]) {
+      LOGW(@"NATS: Cannot send command response - not connected (non-fatal)");
+      return;
+    }
+
+    natsStatus status =
+        natsConnection_Publish(self.conn, [replyTopic UTF8String], responseData.data(),
+                               static_cast<int>(responseData.length()));
+    if (status != NATS_OK) {
+      LOGE(@"NATS: Failed to publish command response to %@: %s (non-fatal)", replyTopic,
+           natsStatus_GetText(status));
+    } else {
+      LOGD(@"NATS: Sent command response to %@ (result: %d)", replyTopic, resultCode);
+    }
+  });
+}
+
+- (void)sendCommandSuccess:(NSString *)message toReplyTopic:(NSString *)replyTopic {
+  [self publishCommandResponse:santa::commands::v1::COMMAND_SUCCESSFUL
+                        output:message
+                  toReplyTopic:replyTopic];
+}
+
+- (void)sendCommandError:(NSString *)errorMessage toReplyTopic:(NSString *)replyTopic {
```

Comment:
I'm not sure I'm finding where `sendCommandSuccess`/`sendCommandError`/`publishCommandResponse:output:toReplyTopic:` are used. Where are these called from (outside of tests)?